web/src/pages/Analytics.tsx

```diff
@@ -0,0 +1,392 @@
+          <Heading fontSize="md">Engagement</Heading>
+          <Text overflowWrap="break-word">
+            Conversion rate with subscribers
+          </Text>
+        </Stack>
+        <NotificationZoneMap />
+      </Stack>
+      <Stack spacing={3}>
+        <Stack spacing={2}>
+          <Heading fontSize="md">Top Performing Customers</Heading>
+          <Text overflowWrap="break-word">
+            Companies with the highest conversion rate
+          </Text>
+        </Stack>
+        <ConversionTable />
+      </Stack>
+    </Stack>
+  );
+};
+
+const StatGrid = () => {
+  const config = useRecoilValue(connectionConfig);
+
+  const overallRateRequests = useSWR(
+    ["overallConversionRateRequests", config],
+    () => overallConversionRate(config, "requests"),
+    { refreshInterval: 1000 }
+  );
+  const overallRatePurchases = useSWR(
+    ["overallConversionRatePurchases", config],
+    () => overallConversionRate(config, "purchases"),
+    { refreshInterval: 1000 }
+  );
+
+  const tableCounts = useSWR(
+    ["analyticsTableCounts", config],
+    () =>
+      estimatedRowCountObj(config, "offers", "subscribers", "notifications"),
+    { refreshInterval: 1000 }
+  );
+
+  if (
+    !tableCounts.data ||
+    !overallRateRequests.data ||
+    !overallRatePurchases.data
+  ) {
+    return <Loader size="small" centered />;
+  }
+
+  return (
+    <Grid gap={2} templateColumns="repeat(2, 1fr)">
+      <StatWrapper
+        statLabel="Offers"
+        statNumber={formatStat(tableCounts.data.offers)}
+        colSpan={2}
+      />
+      <StatWrapper
+        statLabel="Subscribers"
+        statNumber={formatStat(tableCounts.data.subscribers)}
+      />
+      <StatWrapper
+        statLabel="Notifications"
+        statNumber={formatStat(tableCounts.data.notifications)}
+      />
+      <StatWrapper
+        statLabel="Conversion Rate"
+        statNumber={formatPct(overallRateRequests.data?.conversionRate || 0)}
+        helpText="Requests"
+      />
+      <StatWrapper
+        statLabel="Conversion Rate"
+        statNumber={formatPct(overallRatePurchases.data?.conversionRate || 0)}
+        helpText="Purchases"
+      />
+    </Grid>
+  );
+};
+
+const ConversionTable = () => {
+  const config = useRecoilValue(connectionConfig);
+  const [sortColumn, setSortColumn] =
+    React.useState<keyof CustomerMetrics>("conversionRate");
+
+  const metricsTableData = useSWR(
+    ["customerMetrics", config, sortColumn],
+    () => customerMetrics(config, "purchases", sortColumn, 10),
+    { refreshInterval: 1000 }
+  );
+  const activeColor = useColorModeValue("#553ACF", "#CCC3F9");
+  const cellLeftPadding = "10px";
+
+  const getTableBody = () => {
+    if (metricsTableData.isValidating && !metricsTableData.data) {
+      return (
+        <Tr>
+          <Td colSpan={4}>
+            <Loader size="small" centered />
+          </Td>
+        </Tr>
+      );
+    }
+    return metricsTableData.data?.map((c) => (
```

Comment:
Probably we need need to say No data and not a loader.

BCS during validation since data haven't received we show loader. But after fetch is complete and still no data then there will never be data and its not loading anything

I will change that to NO data

Reply:
Added No data comment when there is no Data